### ISSUE TITLE
MD029 is triggered on unordered with numbers in items

### ISSUE THREAD

**Issue by rrauenza** [DavidAnson/markdownlint#451]
I have a bit of a strange use case, perhaps, but I'm doing:

```
#### Header
- 1. item
- 2. item
- 3. item
```
This triggers MD029 even though these should be unordered.

(And the reason for this... I'm using md to store enumerated user stories, but want it parseable by testing tools.  So I want the markdown to have these as lists, but I want them explicitly numbered and never renumbered.)
